server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -823,9 +826,5 @@ pgcolvalueToGValue colVal = case colVal of
   PGValUnknown t -> pure $ G.VString $ G.StringValue t
 
 appendJoinError :: GQRespValue -> GQJoinError -> GQRespValue
-appendJoinError resp err =
-  resp
-    { gqRespData = gqRespData resp
-    , gqRespErrors =
-        pure (gQJoinErrorToValue err : fromMaybe mempty (gqRespErrors resp))
-    }
+appendJoinError resp err = 
+  gqRespErrors <>~ (VB.singleton $ gQJoinErrorToValue err) $ resp
```

Comment:
:eyes:

Reply:
Is this too crazy? It's just `<>` but applied on the value the lens points to. Maybe I should just use the regular `%~` modify operator...

Reply:
Oh no no. This is awesome. That emoji is just an expression of seeing this magic.

Reply:
Ha okay cool, I mean I'm not sure about it myself. But I guess these operators follow a logical pattern of operator-you-know followed by `~` so maybe they are okay.